packages/starlight/utils/navigation.ts

```diff
@@ -410,4 +410,7 @@ function applyPrevNextLinkConfig(
 }
 
 /** Remove the extension from a path. */
-const stripExtension = (path: string) => path.replace(/\.\w+$/, '');
+function stripExtension(path: string) {
+	const periodIndex = path.lastIndexOf('.');
+	return periodIndex < path.length - 1 ? path.slice(0, periodIndex) : path;
```

Comment:
Could comparing to [`-1`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf#return_value) a bit easier to understand?

Reply:
Wait, just having second thought, is this correct? Isn't this always slicing?

Reply:
Ohhhhhh, yes 😅

Probably not breaking tests because this always receives a path like `something.md`, but yeah, should fix the logic to not risk breaking if we ever pass something that doesn’t have an extension.

Reply:
Yeah, did not manage to figure out how to change my approval to request a change but now that you've seen my comment, it's fine ^^

Reply:
Think this might be the best solution, especially given we know this always needs to slice currently:

```suggestion
	return path.slice(0, periodIndex > -1 ? periodIndex: undefined);
```

Reply:
Looks perfect to me and definitely easier to read/understand (and correct) 👍